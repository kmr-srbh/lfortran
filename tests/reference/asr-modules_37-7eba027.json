{
    "basename": "asr-modules_37-7eba027",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/modules_37.f90",
    "infile_hash": "38f6c4643686768ff7e53fc236694c9de56c70b63a8b18bc70765a20",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-modules_37-7eba027.stdout",
    "stdout_hash": "211f1f49f18019afda38c3354a5d0edf2e166a338b3eaf5f1317b1e0",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}