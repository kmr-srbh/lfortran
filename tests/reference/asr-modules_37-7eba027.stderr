Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
[2m  Binary file "[0m[1m[35m$DIR/src/bin/lfortran[39m[0m[2m", in _start[0m
[2m  File "[0m[1m[35m./csu/../csu/libc-start.c[39m[0m[2m", line 360, in __libc_start_main[0m
[2m  File "[0m[1m[35m./csu/../sysdeps/nptl/libc_start_call_main.h[39m[0m[2m", line 58, in __libc_start_call_main[0m
[2m  File "[0m[1m[35m$DIR/src/bin/lfortran.cpp[39m[0m[2m", line 2919, in main[0m
    return main_app(argc, argv);
[2m  File "[0m[1m[35m$DIR/src/bin/lfortran.cpp[39m[0m[2m", line 2738, in main_app(int, char**)[0m
    return emit_asr(arg_file, lfortran_pass_manager,
[2m  File "[0m[1m[35m$DIR/src/bin/lfortran.cpp[39m[0m[2m", line 679, in (anonymous namespace)::emit_asr(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::PassManager&, LCompilers::CompilerOptions&)[0m
    r = fe.get_asr2(input, lm, diagnostics);
[2m  File "[0m[1m[35m$DIR/src/lfortran/fortran_evaluator.cpp[39m[0m[2m", line 304, in LCompilers::FortranEvaluator::get_asr2(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::LocationManager&, LCompilers::diag::Diagnostics&)[0m
    Result<ASR::TranslationUnit_t*> res2 = get_asr3(*ast, diagnostics);
[2m  File "[0m[1m[35m$DIR/src/lfortran/fortran_evaluator.cpp[39m[0m[2m", line 350, in LCompilers::FortranEvaluator::get_asr3(LCompilers::LFortran::AST::TranslationUnit_t&, LCompilers::diag::Diagnostics&)[0m
    compiler_options.symtab_only, compiler_options);
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_to_asr.cpp[39m[0m[2m", line 107, in LCompilers::LFortran::ast_to_asr(Allocator&, LCompilers::LFortran::AST::TranslationUnit_t&, LCompilers::diag::Diagnostics&, LCompilers::SymbolTable*, bool, LCompilers::CompilerOptions&)[0m
    instantiate_types, instantiate_symbols, entry_functions, entry_function_arguments_mapping, data_structure);
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 3890, in LCompilers::LFortran::body_visitor(Allocator&, LCompilers::LFortran::AST::TranslationUnit_t&, LCompilers::diag::Diagnostics&, LCompilers::ASR::asr_t*, LCompilers::CompilerOptions&, std::map<unsigned long, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::ttype_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::ttype_t*> > >, std::less<unsigned long>, std::allocator<std::pair<unsigned long const, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::ttype_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::ttype_t*> > > > > >&, std::map<unsigned long, LCompilers::ASR::symbol_t*, std::less<unsigned long>, std::allocator<std::pair<unsigned long const, LCompilers::ASR::symbol_t*> > >&, std::map<unsigned long, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::less<unsigned long>, std::allocator<std::pair<unsigned long const, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > >&, std::map<unsigned int, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::ttype_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::ttype_t*> > >, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::ttype_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::ttype_t*> > > > > >&, std::map<unsigned int, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::symbol_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::symbol_t*> > >, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::symbol_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::symbol_t*> > > > > >&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<LCompilers::LFortran::AST::stmt_t*, std::allocator<LCompilers::LFortran::AST::stmt_t*> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<LCompilers::LFortran::AST::stmt_t*, std::allocator<LCompilers::LFortran::AST::stmt_t*> > > > >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<LCompilers::LFortran::AST::stmt_t*, std::allocator<LCompilers::LFortran::AST::stmt_t*> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<LCompilers::LFortran::AST::stmt_t*, std::allocator<LCompilers::LFortran::AST::stmt_t*> > > > > > > >&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<int, std::allocator<int> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<int, std::allocator<int> > > > >&, std::vector<LCompilers::ASR::stmt_t*, std::allocator<LCompilers::ASR::stmt_t*> >&)[0m
    b.visit_TranslationUnit(ast);
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 146, in LCompilers::LFortran::BodyVisitor::visit_TranslationUnit(LCompilers::LFortran::AST::TranslationUnit_t const&)[0m
    visit_ast(*x.m_items[i]);
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4884, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::BodyVisitor>::visit_ast(LCompilers::LFortran::AST::ast_t const&)[0m
    void visit_ast(const ast_t &b) { visit_ast_t(b, self()); }
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4841, in void LCompilers::LFortran::AST::visit_ast_t<LCompilers::LFortran::BodyVisitor>(LCompilers::LFortran::AST::ast_t const&, LCompilers::LFortran::BodyVisitor&)[0m
    case astType::mod: { v.visit_mod((const mod_t &)x); return; }
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4887, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::BodyVisitor>::visit_mod(LCompilers::LFortran::AST::mod_t const&)[0m
    void visit_mod(const mod_t &b) { visit_mod_t(b, self()); }
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4471, in void LCompilers::LFortran::AST::visit_mod_t<LCompilers::LFortran::BodyVisitor>(LCompilers::LFortran::AST::mod_t const&, LCompilers::LFortran::BodyVisitor&)[0m
    case modType::Module: { v.visit_Module((const Module_t &)x); return; }
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 1555, in LCompilers::LFortran::BodyVisitor::visit_Module(LCompilers::LFortran::AST::Module_t const&)[0m
    visit_SubmoduleModuleCommon(x);
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 1531, in void LCompilers::LFortran::BodyVisitor::visit_SubmoduleModuleCommon<LCompilers::LFortran::AST::Module_t>(LCompilers::LFortran::AST::Module_t const&)[0m
    visit_program_unit(*x.m_contains[i]);
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4892, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::BodyVisitor>::visit_program_unit(LCompilers::LFortran::AST::program_unit_t const&)[0m
    void visit_program_unit(const program_unit_t &b) { visit_program_unit_t(b, self()); }
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4482, in void LCompilers::LFortran::AST::visit_program_unit_t<LCompilers::LFortran::BodyVisitor>(LCompilers::LFortran::AST::program_unit_t const&, LCompilers::LFortran::BodyVisitor&)[0m
    case program_unitType::Subroutine: { v.visit_Subroutine((const Subroutine_t &)x); return; }
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 1995, in LCompilers::LFortran::BodyVisitor::visit_Subroutine(LCompilers::LFortran::AST::Subroutine_t const&)[0m
    transform_stmts(body, x.n_body, x.m_body);
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 117, in LCompilers::LFortran::BodyVisitor::transform_stmts(LCompilers::Vec<LCompilers::ASR::stmt_t*>&, unsigned long, LCompilers::LFortran::AST::stmt_t**)[0m
    this->visit_stmt(*m_body[i]);
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4930, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::BodyVisitor>::visit_stmt(LCompilers::LFortran::AST::stmt_t const&)[0m
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4616, in void LCompilers::LFortran::AST::visit_stmt_t<LCompilers::LFortran::BodyVisitor>(LCompilers::LFortran::AST::stmt_t const&, LCompilers::LFortran::BodyVisitor&)[0m
    case stmtType::DoLoop: { v.visit_DoLoop((const DoLoop_t &)x); return; }
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 3413, in LCompilers::LFortran::BodyVisitor::visit_DoLoop(LCompilers::LFortran::AST::DoLoop_t const&)[0m
    transform_stmts(body, x.n_body, x.m_body);
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 117, in LCompilers::LFortran::BodyVisitor::transform_stmts(LCompilers::Vec<LCompilers::ASR::stmt_t*>&, unsigned long, LCompilers::LFortran::AST::stmt_t**)[0m
    this->visit_stmt(*m_body[i]);
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4930, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::BodyVisitor>::visit_stmt(LCompilers::LFortran::AST::stmt_t const&)[0m
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4611, in void LCompilers::LFortran::AST::visit_stmt_t<LCompilers::LFortran::BodyVisitor>(LCompilers::LFortran::AST::stmt_t const&, LCompilers::LFortran::BodyVisitor&)[0m
    case stmtType::AssociateBlock: { v.visit_AssociateBlock((const AssociateBlock_t &)x); return; }
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 959, in LCompilers::LFortran::BodyVisitor::visit_AssociateBlock(LCompilers::LFortran::AST::AssociateBlock_t const&)[0m
    transform_stmts(body, x.n_body, x.m_body);
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 117, in LCompilers::LFortran::BodyVisitor::transform_stmts(LCompilers::Vec<LCompilers::ASR::stmt_t*>&, unsigned long, LCompilers::LFortran::AST::stmt_t**)[0m
    this->visit_stmt(*m_body[i]);
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4930, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::BodyVisitor>::visit_stmt(LCompilers::LFortran::AST::stmt_t const&)[0m
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
[2m  File "[0m[1m[35m$DIR/src/lfortran/ast.h[39m[0m[2m", line 4576, in void LCompilers::LFortran::AST::visit_stmt_t<LCompilers::LFortran::BodyVisitor>(LCompilers::LFortran::AST::stmt_t const&, LCompilers::LFortran::BodyVisitor&)[0m
    case stmtType::Assignment: { v.visit_Assignment((const Assignment_t &)x); return; }
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_body_visitor.cpp[39m[0m[2m", line 2480, in LCompilers::LFortran::BodyVisitor::visit_Assignment(LCompilers::LFortran::AST::Assignment_t const&)[0m
    value = adjust_character_length(x, al, value->base.loc, target_type, value);
[2m  File "[0m[1m[35m$DIR/src/lfortran/semantics/ast_common_visitor.h[39m[0m[2m", line 2135, in LCompilers::ASR::expr_t* LCompilers::LFortran::CommonVisitor<LCompilers::LFortran::BodyVisitor>::adjust_character_length<LCompilers::LFortran::AST::Assignment_t>(LCompilers::LFortran::AST::Assignment_t, Allocator&, LCompilers::Location const&, LCompilers::ASR::ttype_t*, LCompilers::ASR::expr_t*, LCompilers::ASR::storage_typeType)[0m
    ASR::Character_t *rhs_type = ASR::down_cast<ASR::Character_t>(
AssertFailed: is_a<T>(*f)
